lib/compress.js

```diff
@@ -1053,9 +1053,7 @@ merge(Compressor.prototype, {
             var val = {};
             for (var i = this.properties.length; --i >= 0;) {
                 var prop = this.properties[i];
-                if (prop instanceof AST_ObjectKeyVal) {
-                    val[prop.key] = ev(prop.value, compressor);
-                }
+                val[prop.key] = ev(prop.value, compressor);
```

Comment:
@alexlamsl Will the following line work in ES5/master?
```
val[ev(prop.key, compressor)] = ev(prop.value, compressor);
```
If so, maybe that's the answer for both branches.